cancat/__init__.py

```diff
@@ -579,7 +579,7 @@ def _isotp_get_msg(self, rx_arbid, start_index=0, service=None, timeout=None):
                     if ord(msg[0]) == 0x7e:  # response for TesterPresent... ignore
                         start_index = msgs[count-1][0] + 1
 
-                    if service is not None:
+                    elif service is not None:
```

Comment:
This logic was wrong - if we get a TesterPresent response, we don't want to keep processing, we want to move on to the next message